src/lib/osUtils/applescript.ts

```diff
@@ -0,0 +1,30 @@
+
+import * as shell from "@tauri-apps/api/shell"
+import { ChildReturnData } from "../types";
+
+export async function execString(command: string): Promise<ChildReturnData> {
+  let args = ['-e', command]
+  console.log("Executing applescript:", args);
+
+  const child = new shell.Command('osascript', args)
+  return new Promise((res) => {
+    let returnData = <ChildReturnData>{ stdout: [], stderr: [] }
```

Comment:
If you need to write something like that, I think following is more idiomatic:
```suggestion
    const returnData: ChildReturnData = { stdout: [], stderr: [] }
```

Reply:
Sure, I will update that. Is there any advantage to typing the variable one way or the other?

Reply:
I honestly didn't test in TypeScript, but generally `const v = <T>{}` is casting and `const v: T = {}` is not, so in second case it will definitely complain if object isn't of the correct shape. I guess your option should also work, but in practice you don't even need it in majority of cases if the function signature is typed, TypeScript will check whether it matches expected interface and should:
1) Infer the type automatically
2) Complain if the type is incorrect

The most often case when you need an explicit type is when you have type `'a' | 'b'` and TypeScript infers wider type `const v: string = 'a'`, but this isn't the case here.

Reply:
ok good to know